### ISSUE TITLE
IRB with binding to another object.

### ISSUE THREAD

**Issue by dekz** [jruby/jruby#384]
Apparently MRI supports multiple IRB sessions and supports the ability to open an IRB session within the context of another object (ie using that objects binding). It seems as if JRuby has problems with this and results in some form of half half mode, but the actual irb context defaults back to a "main" Object.

``` ruby
require 'irb'
obj = Object.new
# Override the objects pretty name formats
obj.instance_eval do
  def inspect; "Test"; end
  def to_s; "Test"; end
end
ARGV.clear
IRB.setup nil

irb = IRB::Irb.new(IRB::WorkSpace.new(obj))
IRB.conf[:MAIN_CONTEXT] = irb.context

# Start the IRB session with this objects binding, passing in nil for file
require 'irb/ext/multi-irb'
IRB.irb nil, obj
```

Now an IRB session should be started using obj's binding. Unforntunately JRuby defaults back to "main" Object, but the prompt has picked up the correct object to use (you can see "Test" in the prompt).

```
irb#1(Test):001:0> self
=> main
```

MRI looks like the following:

```
irb#1(Test):001:0> self
=> Test
```

Tested using MRI 1.9.3-p194 and JRuby 1.7.0 release.

**Comment by MSNexploder** [comment]
Looks like this is working correctly on master

```
jruby --version
jruby 9.0.0.0-SNAPSHOT (2.2.0p0) 2015-01-01 fb9199f Java HotSpot(TM) 64-Bit Server VM 25.25-b02 on 1.8.0_25-b17 +jit [darwin-x86_64]

...

irb#1(Test):001:0> self
=> Test
irb#1(Test):002:0> IRB.conf[:MAIN_CONTEXT].workspace.binding.eval "self"
=> Test
irb#1(Test):003:0> 
```

**Comment by enebo** [comment]
This works in 9k and we will never fix this in 1.7.x as it is winding down....resolving.
